tools/ds-identify

```diff
@@ -1429,11 +1428,15 @@ dscheck_VMware() {
 
     # Activate the VMware datasource only if any of the fields used
     # by the datasource are present in the guestinfo table.
-    if { vmware_rpctool_guestinfo_metadata || \
-         vmware_rpctool_guestinfo_userdata || \
-         vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
-        return "${DS_FOUND}"
-    fi
+    local _val=''
+    for _key in "metadata" "userdata" "vendordata"; do
+      _val="$(eval vmware_rpctool_guestinfo_${_key} 2>&1)"
+      if [ $? -eq 0 ]; then
+          if echo "${_val}" | grep -q "[[:alnum:]]"; then
```

Comment:
Please use the provided example as it is more efficient by not requiring the additional process executions.

Reply:
Please note I updated the above script example to redirect stderr to `/dev/null` to prevent an error message from triggering activation.

Reply:
Actually, we're overthinking this. This isn't what we should be doing. Hang on.

Reply:
We should stick with this:

```shell
    if { vmware_rpctool_guestinfo_metadata || \
          vmware_rpctool_guestinfo_userdata || \
          vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
         return "${DS_FOUND}"
     fi
```

But update:

```shell
vmware_rpctool_guestinfo() {
    local _val=""
    if ! _val="$(vmware-rpctool "info-get guestinfo.${1}")"; then
        return "${?}"
    fi
    return [ -n "${val/ /}" ]
}

vmware_rpctool_guestinfo_metadata() {
    vmware_rpctool_guestinfo "metadata"
}

vmware_rpctool_guestinfo_userdata() {
    vmware_rpctool_guestinfo "userdata"
}

vmware_rpctool_guestinfo_vendordata() {
     vmware_rpctool_guestinfo "vendordata"
}
```

Reply:
Okay, as @sshedi just pointed out on Slack, string replacement is not POSX-shell compliant. Okay, here's how we optimize this then:

```shell
    # Activate the VMware datasource only if any of the fields used
    # by the datasource are present in the guestinfo table.
    if { vmware_rpctool_guestinfo_metadata && \
         vmware_rpctool_guestinfo_userdata && \
         vmware_rpctool_guestinfo_vendordata; } 2>/dev/null | \
         grep -q "[[:alnum:]]"; then
        return "${DS_FOUND}"
    fi
```

Reply:
We also just realized that the above example's best case will still require three RPC calls. I think we'd rather have three calls to `grep` than three RPC calls, so:

```shell
vmware_rpctool_guestinfo() {
    vmware-rpctool "info-get guestinfo.${1}" 2>/dev/null | grep "[[:alnum:]]"
}

vmware_rpctool_guestinfo_metadata() {
    vmware_rpctool_guestinfo "metadata"
}

vmware_rpctool_guestinfo_userdata() {
    vmware_rpctool_guestinfo "userdata"
}

vmware_rpctool_guestinfo_vendordata() {
     vmware_rpctool_guestinfo "vendordata"
}
```

Then the following can still work:

```shell
    if { vmware_rpctool_guestinfo_metadata || \
         vmware_rpctool_guestinfo_userdata || \
         vmware_rpctool_guestinfo_vendordata; } >/dev/null 2>&1; then
         return "${DS_FOUND}"
     fi
```

Reply:
Just verified the following patch still passes the unit tests:

```diff
diff --git a/tools/ds-identify b/tools/ds-identify
index 234ffa81..9efb647f 100755
--- a/tools/ds-identify
+++ b/tools/ds-identify
@@ -1384,16 +1384,20 @@ vmware_has_rpctool() {
     command -v vmware-rpctool >/dev/null 2>&1
 }
 
+vmware_rpctool_guestinfo() {
+    vmware-rpctool "info-get guestinfo.${1}" 2>/dev/null | grep "[[:alnum:]]"
+}
+
 vmware_rpctool_guestinfo_metadata() {
-    vmware-rpctool "info-get guestinfo.metadata"
+    vmware_rpctool_guestinfo "metadata"
 }
 
 vmware_rpctool_guestinfo_userdata() {
-    vmware-rpctool "info-get guestinfo.userdata"
+    vmware_rpctool_guestinfo "userdata"
 }
 
 vmware_rpctool_guestinfo_vendordata() {
-    vmware-rpctool "info-get guestinfo.vendordata"
+     vmware_rpctool_guestinfo "vendordata"
 }
 
 dscheck_VMware() {
```